src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionEndpoint.cs

```diff
@@ -34,7 +34,7 @@ internal class CodeActionEndpoint : ICodeActionHandler
             IEnumerable<RazorCodeActionProvider> providers,
             ForegroundDispatcher foregroundDispatcher,
             DocumentResolver documentResolver,
-            ILanguageServer languageServer)
```

Comment:
is ILanguageServer not registered as a service anymore?

Reply:
No, the problem is that O#'s new DI system introduces a recursive resolution here trying to create an ICodeActionHander to put in ILanguageServer, which goes in ICodeACtionHandler.